pandas/_libs/tslibs/period.pyx

```diff
@@ -1756,18 +1756,22 @@ cdef class _Period:
         -------
         Timestamp
         """
-        if freq is not None:
-            freq = self._maybe_convert_freq(freq)
         how = validate_end_alias(how)
 
         end = how == 'E'
         if end:
+            if freq == "B":
```

Comment:
Are there no other freqs for which you need to do a similar adjustment? (like the other business-related freqs?)